ui/browser/views/browser.jsx

```diff
@@ -40,17 +39,25 @@ import '../../shared/web-view';
 class BrowserWindow extends Component {
   componentDidMount() {
     attachListeners(this.props);
+
+    this.props.ipcRenderer.on('profile-diff', (event, args) => {
+      this.props.dispatch(args);
+    });
   }
 
   render() {
     const { dispatch, pages, currentPageIndex, pageOrder, ipcRenderer,
+            profile,
             pageAreaVisible } = this.props;
     const navBack = e => getCurrentWebView(e.target.ownerDocument).goBack();
     const navForward = e => getCurrentWebView(e.target.ownerDocument).goForward();
     const navRefresh = e => getCurrentWebView(e.target.ownerDocument).reload();
     const openMenu = () => menuBrowser(dispatch);
-    const bookmark = (title, url) => bookmarkAction(title, url, dispatch);
-    const unbookmark = (url) => unbookmarkAction(url, dispatch);
+    const isBookmarked = (url) => profile.bookmarks.has(url);
```

Comment:
On Thu, Apr 14, 2016 at 6:06 PM, Jordan Santell <notifications@github.com>
wrote:

> In ui/browser/views/browser.jsx
> <https://github.com/mozilla/tofino/pull/130#discussion_r59815668>:
>
> >              pageAreaVisible } = this.props;
> >      const navBack = e => getCurrentWebView(e.target.ownerDocument).goBack();
> >      const navForward = e => getCurrentWebView(e.target.ownerDocument).goForward();
> >      const navRefresh = e => getCurrentWebView(e.target.ownerDocument).reload();
> >      const openMenu = () => menuBrowser(dispatch);
> > -    const bookmark = (title, url) => bookmarkAction(title, url, dispatch);
> > -    const unbookmark = (url) => unbookmarkAction(url, dispatch);
> > +    const isBookmarked = (url) => profile.bookmarks.has(url);
>
> isBookmarked is a part of the Page object, we should have one or the other
>
Oh, right, I removed consumers but not the field.  I'm happy to update the
`Page` field, or to re-think the abstraction.  It's weird to me that the
current page "leaks up" the UI hierarchy to touch the navbar, but I
followed suite from what was here.  Do you have a strong preference?  (I
don.t)